network/src/commonMain/kotlin/com/wire/kalium/network/api/v4/authenticated/NotificationApiV4.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.network.api.v4.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.AuthenticatedWebSocketClient
+import com.wire.kalium.network.api.base.authenticated.notification.NotificationResponse
+import com.wire.kalium.network.api.v3.authenticated.NotificationApiV3
+import com.wire.kalium.network.tools.ServerConfigDTO
+import com.wire.kalium.network.utils.NetworkResponse
+import com.wire.kalium.network.utils.wrapKaliumResponse
+import io.ktor.client.request.get
+import io.ktor.client.request.parameter
+
+internal open class NotificationApiV4 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient,
+    authenticatedWebSocketClient: AuthenticatedWebSocketClient,
+    serverLinks: ServerConfigDTO.Links
+) : NotificationApiV3(authenticatedNetworkClient, authenticatedWebSocketClient, serverLinks) {
+
+    private val httpClient get() = authenticatedNetworkClient.httpClient
+
+    override suspend fun notificationsCall(
+        querySize: Int,
+        queryClient: String,
+        querySince: String?
+    ): NetworkResponse<NotificationResponse> = wrapKaliumResponse {
+        // Pretty much the same V0 request, but without the 404 overwrite
+        httpClient.get(V0.PATH_NOTIFICATIONS) {
```

Comment:
isn't better to have the variables defined for each api version class? here we using the V0 variables.

Reply:
Yes, but I'm not implementing anything new here. I'm adding api V4 by adding the necessary scaffolding. I'm not touching the implementation.